src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -34,14 +40,25 @@ public class ServiceBusJmsConnectionFactory extends JNDIStorable implements Conn
     // JNDI property names
     private static final String CONNECTION_STRING_PROPERTY = "connectionString";
     private static final String CLIENT_ID_PROPERTY = "clientId";
-    
+    private static final String AUDIENCE = "https://servicebus.azure.net/.default";
+    private static final String AUTHORITY = "https://login.microsoftonline.com/%s/";
     private static final int MaxCustomUserAgentLength = 128;
+    
     private final ServiceBusJmsConnectionFactorySettings settings;
     private volatile boolean initialized;
     private JmsConnectionFactory factory;
     private ConnectionStringBuilder builder;
     private String customUserAgent;
     
+    private String tenantId; 
+    private String aadClientId; 
+    private String clientSecret;
+    private String keyName;
+    private String keyValue;
```

Comment:
@vinaysurya don`t you think that all of this should be moved to some kind of generic class TokenCredential with different implementations ClientSecretCredential, AzureNamedKeyCredential, AzureSasCredential? It will be cleaner and add more flexibility.

Code examples:
[TokenCredential field](https://github.com/Azure/azure-sdk-for-java/blob/c3dd93dea3887a1a727cda76273b6d40c83a97bc/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java#L219)
[Construct factory using connection string](https://github.com/Azure/azure-sdk-for-java/blob/c3dd93dea3887a1a727cda76273b6d40c83a97bc/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java#L321)
[Construct factory using token credentials](https://github.com/Azure/azure-sdk-for-java/blob/c3dd93dea3887a1a727cda76273b6d40c83a97bc/sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java#L415)

More examples:
[TokenCredential class](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenCredential.java)
[ClientSecretCredential](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientSecretCredential.java)
[Construct factory using connection string](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/servicebus/azure-messaging-servicebus#create-service-bus-clients-using-a-connection-string)
[Construct factory using active directory](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/servicebus/azure-messaging-servicebus#create-a-service-bus-client-using-microsoft-identity-platform-formerly-azure-active-directory)

Reply:
Yes since this was made only thinking on app service and not msi too I think using 
TokenCredential credential = new DefaultAzureCredentialBuilder()
            .build();

will address both.